sympy/physics/mechanics/_system.py

```diff
@@ -0,0 +1,937 @@
+        """Retrieve a joint from the system by name.
+
+        Parameters
+        ==========
+
+        name : str
+            The name of the joint to retrieve.
+
+        Returns
+        =======
+
+        subclass of Joint
+            The joint with the given name, or None if no such joint exists.
+
+        """
+        for joint in self._joints:
+            if joint.name == name:
+                return joint
+
+    def _form_eoms(self):
+        return self.form_eoms()
+
+    def form_eoms(self, eom_method=KanesMethod):
+        """Form the equations of motion of the system.
+
+        Parameters
+        ==========
+
+        eom_method : class
+            Backend class to be used for forming the equations of motion. The
+            default is ``KanesMethod``.
+
+        Returns
+        ========
+
+        Matrix
+            Vector of equations of motions.
+
+        Examples
+        ========
+
+        As the ``_system.py`` module is experimental, it is not yet part of the
+        ``sympy.physics.mechanics`` namespace. ``System`` must therefore be
+        imported directly from the ``sympy.physics.mechanics._system`` module.
+
+        >>> from sympy.physics.mechanics._system import System
+
+        This is a simple example for a one degree of freedom translational
+        spring-mass-damper.
+
+        >>> from sympy import S, symbols
+        >>> from sympy.physics.mechanics import (
+        ...     LagrangesMethod, dynamicsymbols, PrismaticJoint, Particle,
+        ...     RigidBody)
+        >>> q = dynamicsymbols('q')
+        >>> qd = dynamicsymbols('q', 1)
+        >>> m, k, b = symbols('m k b')
+        >>> wall = RigidBody('W')
+        >>> system = System.from_newtonian(wall)
+        >>> bob = Particle('P', mass=m)
+        >>> bob.potential_energy = S.Half * k * q**2
+        >>> system.add_joints(PrismaticJoint('J', wall, bob, q, qd))
+        >>> system.apply_force(bob, b * qd * system.x, reaction_point=wall)
+        >>> system.form_eoms(LagrangesMethod)
+        Matrix([[-b*Derivative(q(t), t) + k*q(t) + m*Derivative(q(t), (t, 2))]])
+
+        We can also solve for the states using the 'rhs' method.
+
+        >>> system.rhs()
+        Matrix([
+        [               Derivative(q(t), t)],
+        [(b*Derivative(q(t), t) - k*q(t))/m]])
+
+        """
+        # KanesMethod does not accept empty iterables
+        loads = self.loads if self.loads else None
+        if issubclass(eom_method, KanesMethod):
+            self._eom_method = KanesMethod(
+                self.frame, self.q_ind, self.u_ind, kd_eqs=self.kdes,
+                q_dependent=self.q_dep, u_dependent=self.u_dep,
+                configuration_constraints=self.holonomic_constraints,
+                velocity_constraints=self.holonomic_constraints.diff(
+                    dynamicsymbols._t).col_join(self.nonholonomic_constraints),
+                forcelist=loads, bodies=self.bodies,
+                explicit_kinematics=False)
```

Comment:
Thoughts on splitting these one kwarg per line? I found this very very difficult to read.

Reply:
Will assign the velocity constraints to a separate variable, as that is probably the part which makes this more difficult to read.

Reply:
Addressed in 9d9a8c1cc3532150f0e4764d9c5a6016cc80a939